src/pxMatrix4T.h

```diff
@@ -253,43 +253,24 @@ void multiply(FloatT* m, FloatT* n)
     out[6] = a12 * c - a02 * s;
     out[7] = a13 * c - a03 * s;
   }
-  
+
   void scale(FloatT sx, FloatT sy) 
-  {   
+  {
     FloatT *out = mValues;
-    FloatT *a = mValues;
-    
-    out[0] = a[0] * sx;
-    out[1] = a[1] * sx;
-    out[2] = a[2] * sx;
-    out[3] = a[3] * sx;
-    out[4] = a[4] * sy;
-    out[5] = a[5] * sy;
-    out[6] = a[6] * sy;
-    out[7] = a[7] * sy;
+    FloatT *a   = mValues;
+
+    out[0]  = a[0]  * sx;
+    out[5]  = a[5]  * sy;
   }
 
   void scale(FloatT sx, FloatT sy, FloatT sz) 
-  {  
+  {
     FloatT *out = mValues;
-    FloatT *a = mValues;
-    
-    out[0] = a[0] * sx;
-    out[1] = a[1] * sx;
-    out[2] = a[2] * sx;
-    out[3] = a[3] * sx;
-    out[4] = a[4] * sy;
-    out[5] = a[5] * sy;
-    out[6] = a[6] * sy;
-    out[7] = a[7] * sy;
-    
-    if (sz != 1.0)
-    {
-      out[8] = a[8] * sz;
-      out[9] = a[9] * sz;
-      out[10] = a[10] * sz;
-      out[11] = a[11] * sz;
-    }
+    FloatT *a   = mValues;
+
+    out[0]  = a[0]  * sx;
+    out[5]  = a[5]  * sy;
+    out[10] = a[10] * sz;
   }
```

Comment:
The changes to scale() will break scaling.

Reply:
Committed changes suggested ... CI build pending.